Use a problem as the intro for newtype pattern (#153)
rust-unofficial/patterns: patterns/newtype.md

# Newtype

What if in some cases we want a type to behave similar to another type but
still different? What if we want to enforce some behavior at compile time

> which using only type aliases is not enough? For example, have a custom

Reviewer: ```suggestion
which using only type aliases is not enough? For example, having a custom
```

> What if in some cases we want a type to behave similar to another type but
> still different? What if we want to enforce some behavior at compile time
> which using only type aliases is not enough? For example, have a custom
> `Display` implementation for `String` for security reasons (password).

Reviewer: ```suggestion
What if in some cases we want a type to behave similar to another type or 
enforce some behaviour at compile time where using only type aliases would 
not be enough? 

For example, if you want to create a custom `Display` implementation for `String` 
due to security considerations (e.g. passwords).
```

Author: The separation looks good but the word "you" does not look that good for me. Either way I just applied it.

> We could use the newtype pattern to provide type safety and encapsulation.

Reviewer: ```suggestion
For aforementioned and such cases we could use the `Newtype` pattern to provide __type safety and encapsulation__.
```

Author: ```suggestion
For such cases we could use the `Newtype` pattern to provide __type safety__ and __encapsulation__.
```

aforementioned is a hard word, I think it is better to remove that.

# Newtype

What if in some cases we want a type to behave similar to another type or 
enforce some behaviour at compile time where using only type aliases would 
not be enough? 

> For example, if you want to create a custom `Display` implementation for `String` 

Reviewer: ```suggestion
For example, if we want to create a custom `Display` implementation for `String` 
```

## See also

- [Advanced Types in the book](https://doc.rust-lang.org/book/ch19-04-advanced-types.html?highlight=newtype#using-the-newtype-pattern-for-type-safety-and-abstraction)

> - [Newtypes in the style guide](https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html)

Author: The style guide is outdated or not maintained IIRC.

Reviewer: Sad, but true.

```suggestion
```
